utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt

```diff
@@ -0,0 +1,23 @@
+package pm.gnosis.tests.utils
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.runner.Description
+
+@ExperimentalCoroutinesApi
+class TestLifecycleRule : InstantTaskExecutorRule() {
+
+    override fun starting(description: Description?) {
+        super.starting(description)
+        Dispatchers.setMain(TestCoroutineDispatcher())
+    }
+
+    override fun finished(description: Description?) {
+        Dispatchers.resetMain()
+        super.finished(description)
+    }

```

Comment:
This file as well as `MainCoroutineScopeRule` are forcing the use of experimental API which I am not sure it is advisable for library dependency. Here I suppose we should aim rather for stability?

Reply:
But this does not become part of any library that uses it? It is just used for testing, is it?

Reply:
Well yeah that's true, this should be a `testImplementation` dependency somewhere. Now that you mention it, I guess this would have to be purposefully imported by someone anyhow so there is no need to worry about that.